app/org/sagebionetworks/bridge/services/IntentService.java

```diff
@@ -114,23 +121,29 @@ public void submitIntentToParticipate(IntentToParticipate intent) {
         }
     }
     
-    public void registerIntentToParticipate(Study study, StudyParticipant participant) {
-        Phone phone = participant.getPhone();
+    public boolean registerIntentToParticipate(Study study, Account account) {
+        Phone phone = account.getPhone();
         // Somehow, this is being called but the user has no phone number.
         if (phone == null) {
-            return;
+            return false;
         }
-        
+        boolean consentsUpdated = false;
+        StudyParticipant participant = null;
         List<Subpopulation> subpops = subpopService.getSubpopulations(study.getStudyIdentifier());
         for (Subpopulation subpop : subpops) {
             CacheKey cacheKey = CacheKey.itp(subpop.getGuid(), study.getStudyIdentifier(), phone);
             IntentToParticipate intent = cacheProvider.getObject(cacheKey, IntentToParticipate.class);
             if (intent != null) {
+                if (participant == null) {
+                    participant = participantService.getParticipant(study, account.getId(), true);
+                }
```

Comment:
Is getParticipant() a particularly expensive call? If it's not, it might be simpler to call getParticipant() before the loop, instead of having these extra if statements.

Reply:
Yes, it's expensive here because we're loading all the consent records and we do use them in the consentToResearch call. I would be careful not to load it unless it's really necessary.